### ISSUE TITLE
Loss function inconsistent with the article

### ISSUE THREAD

**Issue by CharlesNord** [Prinsphield/ELEGANT#11]
https://github.com/Prinsphield/ELEGANT/blob/6a36225a958ceee95419d7f0a098dae5c749b7f7/ELEGANT.py#L203
In your code , the loss function of discriminator is not the same as in the paper. The weight of losses from two discriminators are not equal, is this setting important and could lead to a better performance?

**Comment by Prinsphield** [comment]
For simplicity, the paper neglects the coefficient in each loss term. Because the overall design is towards the right way, so it will work finally.

Actually, I have not tried other coefficients. It may lead to better performance.  You may adjust the coefficient accordingly when the total number of attributes or the training attributes get changed in order to obtain better performance.

**Comment by CharlesNord** [comment]
> For simplicity, the paper neglects the coefficient in each loss term. Because the overall design is towards the right way, so it will work finally.
> 
> Actually, I have not tried other coefficients. It may lead to better performance. You may adjust the coefficient accordingly when the total number of attributes or the training attributes get changed in order to obtain better performance.

I get it, thank you very much for your reply
